api/app/routers/url.py

```diff
@@ -21,36 +22,44 @@
 
 
 @router.get("/", response_model=List[schemas.Url])
-async def get_my_urls(username: str, db: Session = Depends(get_db)):
-    return crud.get_user_urls(db=db, username=username)
+async def get_my_urls(db: Session = Depends(get_db), user=Depends(get_current_user)):
+    return crud.get_user_urls(db=db, username=user["login"])
 
 
 @router.put("/", response_model=schemas.Url, status_code=status.HTTP_201_CREATED)
-async def put_url(url: schemas.UrlCreate, db: Session = Depends(get_db)):
+async def put_url(url: schemas.UrlCreate, db: Session = Depends(get_db), user=Depends(get_current_user)):
```

Comment:
`owner_username` is still mandatory if `schemas.UrlCreate` (herited from `schemas.UrlBase`). I think that moving it to the `UrlBase` should do the job :)

Reply:
`owner_username` is already in `UrlBase` so I am not sure to understand where do you want to move it.
Maybe we could make it optional if we want to allow non registered users to shorten urls.
Otherwise we could create a `RequestUrl(BaseModel)` ?

Reply:
Sorry edited 👼

Reply:
Fixed in the new version.